sqlalchemy-stubs/ext/asyncio/engine.pyi

```diff
@@ -1,103 +1,128 @@
 from typing import Any
 from typing import Callable
+from typing import Generator
 from typing import Mapping
+from typing import NoReturn
 from typing import Optional
+from typing import TypeVar
 
-from .base import ProxyComparable as ProxyComparable
-from .base import StartableContext as StartableContext
-from .result import AsyncResult as AsyncResult
-from ... import exc as exc
-from ... import util as util
-from ...engine import Result as Result
-from ...engine import Transaction as Transaction
-from ...future import Connection as Connection
-from ...future import Engine as Engine
-from ...sql import Executable as Executable
-from ...util.concurrency import greenlet_spawn as greenlet_spawn
+from .base import ProxyComparable
+from .base import StartableContext
+from .result import AsyncResult
+from ...engine import Result
+from ...engine import Transaction
+from ...future import Connection
+from ...future import Engine
+from ...sql import Executable
 
-def create_async_engine(*arg: Any, **kw: Any): ...
+_TAsyncConnection = TypeVar("_TAsyncConnection", bound=AsyncConnection)
+_TAsyncTransaction = TypeVar("_TAsyncTransaction", bound=AsyncTransaction)
+
+def create_async_engine(*arg: Any, **kw: Any) -> AsyncEngine: ...
 
 class AsyncConnectable: ...
 
 class AsyncConnection(ProxyComparable, StartableContext, AsyncConnectable):
-    engine: Any = ...
+    # copied from future.Connection via create_proxy_methods
+    @property
+    def closed(self) -> bool: ...
+    @property
+    def invalidated(self) -> bool: ...
+    dialect: Any
+    @property
+    def default_isolation_level(self) -> Any: ...
+    # end copied
```

Comment:
I've also asked here https://github.com/sqlalchemy/sqlalchemy2-stubs/pull/52#discussion_r611037575, can't we create a superclass just in the typing for these?

Reply:
Yeah, I meant to reply there. I think that's a good idea and I'll go ahead and implement that.

Reply:
As I said in the other PR: thinking about this a bit more, I think we (me, most likely) should implement this change after all of the current PRs are landed so as to reduce changes between files right now.

Reply:
ref https://github.com/sqlalchemy/sqlalchemy2-stubs/issues/61